coverage/html.py

```diff
@@ -317,7 +317,9 @@ def report(self, morfs: Iterable[TMorf] | None) -> float:
                 file_be_gone(os.path.join(self.directory, ftr.html_filename))
 
         if not have_data:
-            raise NoDataError("No data to report.")
+            raise DataFileOrDirectoryNotFoundError.new_for_data_file_or_directory(
+                os.path.dirname(self.coverage.get_data().base_filename())
```

Comment:
I have to think through whether the message here is correct.  Is it always that the file doesn't exist?

Reply:
It's good that the exceptions centralize the messages, but everywhere this exception is created it uses the same fairly complex expression.  Can we centralize that as well?